lib/elixir/lib/kernel.ex

```diff
@@ -3174,36 +3252,44 @@ defmodule Kernel do
         case ensure_evaled(list, {0, []}, expand) do
           {[head | tail], {_, []}} ->
             in_var(in_module?, left, &in_list(&1, head, tail, expand, list, in_module?))
+
           {[head | tail], {_, vars_values}} ->
             {vars, values} = :lists.unzip(:lists.reverse(vars_values))
+
             quote do
               {unquote_splicing(vars)} = {unquote_splicing(values)}
-              unquote(in_var(in_module?, left, &in_list(&1, head, tail, expand, list, in_module?)))
```

Comment:
Please extra `&in_list(&1, head, tail, expand, list, in_module?)` to a variable outside of the `quote` (two lines above) and then use it here.